### PR TITLE
MULE-19055: SampleDataProvider validator is not validating generic

### PR DESCRIPTION
**Pull request by marianogonzalez**

### CODE REVIEW

**Review by ndinu — DISMISSED** [review]

**Review by ndinu — DISMISSED** [review]

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/validation/SampleDataModelValidator.java [hunk 1]
```diff
@@ -144,6 +157,38 @@ private void validateResolver(ConnectableComponentModel model,
...
+        .map(this::asString)
+        .collect(joining(",")) + ">";
+  }
+
+  private String asString(Type type) {
+    if (type instanceof ParameterizedTypeImpl) {
+      ParameterizedTypeImpl parameterizedType = (ParameterizedTypeImpl) type;
+      return parameterizedType.getRawType().getName() + asGenericSignature(asList(parameterizedType.getActualTypeArguments()));
+    } else {
+      return type != null ? type.getTypeName() : Object.class.getName();
```
Comment (ndinu) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/validation/SampleDataModelValidator.java:188]: Maybe when the type is null, and is the generic of the SampleDataProvider, we should add an error that states that the SampleDataProviderGenerics must be explicit.

Comment (marianogonzalez) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/validation/SampleDataModelValidator.java:188]: This is for cases in which one of the provider's generic types is a Map or some other parameterized type where we cannot take it upon ourselves to demand proper use of those clases

**Review by ndinu — APPROVED** [review]

**Comment by marianogonzalez** [comment]
--validate

**Comment by marianogonzalez** [comment]
--validate

**Comment by marianogonzalez** [comment]
--validate

**Comment by marianogonzalez** [comment]
--validate

**Comment by marianogonzalez** [comment]
--validate

**Pull request merged**